src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -32,6 +34,29 @@
     private static Bundle gCachedExtras = null;
     private static boolean gForeground = false;
 
+    @Override
+    public void initialize(CordovaInterface cordova, CordovaWebView webView) {
+        super.initialize(cordova, webView);
+        gForeground = true;
+
+        if (isAmazonDevice()) {
```

Comment:
I'd flip this if as it is more likely that it will be a Google Device not an Amazon one.